rudder-core/src/main/scala/com/normation/rudder/domain/nodes/Node.scala

```diff
@@ -220,12 +198,12 @@ object CompareProperties {
                        oldPropsMap.get(newProp.name) match {
                          case None =>
                            Full(updateOrRemoveProp(newProp))
-                         case Some(oldProp@NodeProperty(name, value, provider, rights)) => rights match {
-                             case Some(ReadOnly) if(!same(newProp.provider, provider)) =>
-                               Failure(s"You are trying to update the following property which is owned by an other provider with 'read-only' mode set: '${name}'")
-                             case _ =>
+                         case Some(oldProp@NodeProperty(name, value, provider)) =>
+                             if(same(newProp.provider, provider)) {
                                Full(updateOrRemoveProp(newProp))
-                           }
+                             } else {
+                               Failure(s"You are trying to update the following property which is owned by an other provider, '${provider.map(_.value).getOrElse("")}': '${name}'")
```

Comment:
Maybe it would be interesting to state here which provider you are  we only display the current provider )